src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -187,43 +240,56 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
         };
 
         // looping through the fields on DataImport__c
-        for (Schema.DescribeFieldResult fieldResult : fieldMap.values()) {
-            String dataImportFieldApiName = UTIL_Namespace.StrTokenRemoveNSPrefix(fieldResult.getName());
+        for (String fieldName : sortedFieldNames) {
+            DescribeFieldResult dfr = fieldMap.get(fieldName);
+            String dataImportFieldApiName = UTIL_Namespace.StrTokenRemoveNSPrefix(dfr.getName());
 
-            String dataImportFieldMappedObject = BDI_DataImportService.getTargetObject(fieldResult.getName());
+            String dataImportFieldMappedObject = BDI_DataImportService.getTargetObject(dfr.getName());
             Boolean fieldAllowed = allowedObjects.contains(dataImportFieldMappedObject)
-                    && allowedDataTypes.contains(fieldResult.getType().name())
+                    && allowedDataTypes.contains(dfr.getType().name())
                     && !bannedFields.contains(dataImportFieldApiName)
-                    && fieldResult.getInlineHelpText() != null;
+                    && dfr.getInlineHelpText() != null
+                    && checkFLS(dfr);
 
             if (fieldAllowed) {
+                dfrs.add(dfr);
+            }
+        }
 
-                BGEField field = new BGEField();
-
-                // TODO: make these localized with schema describes
-                field.sObjectName = dataImportFieldMappedObject;
-
-                // TODO: pull these labels from target fields' schema based on help text mapping
-                Boolean fieldHasSpacesAndObjName = fieldResult.getLabel().contains(' ') && fieldResult.getLabel().split(' ',2)[0] == field.sObjectName;
-                field.label = (fieldHasSpacesAndObjName) ? fieldResult.getLabel().split(' ',2)[1] : fieldResult.getLabel();
-                field.name = fieldResult.getName();
-
-                String unnamespacedField = UTIL_Namespace.StrTokenRemoveNSPrefix(field.name);
-                if (systemRequiredFields.contains(unnamespacedField)) {
-                    field.systemRequired = true;
-                }
-
-                String rawFieldType = fieldResult.getType().name().toLowerCase();
-                field.type = getLightningFieldType(rawFieldType);
-                if (rawFieldType == 'currency') {
-                    field.formatter = 'currency';
-                }
+        return dfrs;
+    }
 
-                field.options = getPicklistOptions(field.name);
+    /*******************************************************************************************************
+    * @description Returns the relevant Batch or Template fields for soql
+    * @return list of field API names
+    */
+    public static List<String> getRecordFieldNames(String sObjectName) {
+
+        String descriptionField = (sObjectName == 'DataImportBatch__c') ? 'Batch_Description__c' : 'Description__c';
+
+        List<String> fields = new List<String>{
```

Comment:
I'm confused about "fields" ... at very least can we name this variable something more descriptive to suggest the purpose of the list?